Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/RunScript.java

```diff
@@ -12,48 +13,53 @@
  */
 public class RunScript extends CommandGroup {
 
-    public RunScript(String scriptName) {
-    		ArrayList<String[]> script = Robot.autoScripts.getOrDefault(scriptName, new ArrayList<String[]>());
-    		
-    		outerloop:
-    		for(String[] cmd : script) {
-    			String cmdName = cmd[0];
-    			String cmdArgs = cmd[1];
-    			
-    			switch (cmdName) {
-	    			case "moveto":
-	    				addSequential(new AutoMoveTo(cmdArgs.split(" ")));
-	    				break;
-	    			case "turn":
-	    				addSequential(new AutoTurn(Double.parseDouble(cmdArgs)));
-	    				break;
-	    			case "move":
-	    				addSequential(new AutoMove(Double.parseDouble(cmdArgs)));
-	    				break;
-	    			case "switch":
-	    				addSequential(new EjectToSwitch());
-	    				break;
-	    			case "scale":
-	    				addSequential(new EjectToScale());
-	    				break;
-	    			case "exchange":
-	    				addSequential(new EjectToExchange());
-	    				break;
-	    			case "wait":
-	    				addSequential(new WaitCommand(Double.parseDouble(cmdArgs)));
-	    				break;
-	    			case "intake":
-	    				addSequential(new IntakeCube());
-	    				break;
-	    			case "jump":
-	    				addSequential(new RunScript(cmdArgs));
-	    				break;
-	    			case "end":
-	    				break outerloop;
-	    			default:
-	    				// this should never happen since AutoUtils already validates the script.
-	    				System.err.println("[ERROR] `" + cmdName + "` is not a valid command name.");
-    			}
-    		}
-    }
+	public RunScript(String scriptName) {
+		ArrayList<String[]> script = Robot.autoScripts.getOrDefault(scriptName, new ArrayList<String[]>());
+
+		outerloop: for (String[] cmd : script) {
+			String cmdName = cmd[0];
+			String cmdArgs = cmd[1];
+
+			switch (cmdName) {
+			case "moveto":
+				addSequential(new AutoMoveTo(cmdArgs.split(" ")));
+				break;
+			case "turn":
+				double rotation = Double.parseDouble(cmdArgs);
+				addSequential(new PIDTurn(rotation, Robot.dt, Robot.sd, Robot.dt.getGyro()));
+				AutoUtils.position.setRot(rotation);
```

Comment:
This code assumes that PIDTurn puts us exactly where we've requested. That is not a valid assumption and will result in accumulating errors.